django/db/backends/mysql/features.py

```diff
@@ -60,6 +60,8 @@ class DatabaseFeatures(BaseDatabaseFeatures):
     order_by_nulls_first = True
     supports_logical_xor = True
 
+    can_use_autoincrement_cols_in_check_constraints = False
```

Comment:
A separate feature flag is not necessary. We can add crashing tests to the `DatabaseFeatures.django_test_skips` on MySQL. However, we don't have to skip tests at all on MySQL when we're not really creating a constraint, just testing that validation works, check out the following comments.